feat!: Add big int opcodes (without implementation)

guipublic opened the pull request:
Adds the biginteger opcode skeleton https://github.com/noir-lang/noir/issues/4040

The PR adds ACIR opcodes for bigint operations. It does not provide any implantation of the opcodes, neither in BB, in the solver or in Brillig.

guipublic committed:
Add big int opcodes (without implementation)

guipublic committed:
Merge branch 'master' into gd/bigint_opcode

guipublic committed:
Use id to identify bigints between Noir and BB

guipublic committed:
Merge branch 'master' into gd/bigint_opcode

guipublic committed:
fix build

kevaundray commented:
Linking to https://github.com/noir-lang/noir/issues/4040 -- this PR doesn't fully resolve it, so lets link other PRs to this issue still!

guipublic committed:
chore: add unit test for BB big integer opcode implementation (#4177)

The unit test is just creating big int constraints and should be updated
when the constraints are implemented.

kevaundray committed:
Merge branch 'master' into gd/bigint_opcode

guipublic committed:
Merge branch 'master' into gd/bigint_opcode

guipublic committed:
fix build issue

guipublic committed:
fix the build (2)
